### DISCUSSION TITLE
How to identify the frame of a MQTT message?

### DISCUSSION THREAD

**Discussion by londi** [meatpiHQ/wican-fw#31] [Q&A]
Hi There
I try to decode the mqtt messages received by my wican adapter in python.
For example:
`wican/xxxxxx/can/rx payload: b'{"bus":"0","type":"rx","ts":20303,"frame":[
{"id":222,"dlc":6,"rtr":false,"extd":false,"data":[28,41,151,208,5,219,0,0]}
]}'
`
Can you help me how to identify which can bus frame this data represents?
I tried to decode it with this [tool](https://www.csselectronics.com/pages/obd2-pid-table-on-board-diagnostics-j1979) but I can't figure it out.

I want to learn how to interpret it so that I can then determine the other frames, e.g. the vehicle speed.

**Edit**: If I send a _2016_ tx frame via mqtt, the meatpi does not responde with a _2024_ frame. (As explained in the docs)

Thank you in advance
Londi

**Comment by meatpiHQ** [comment]
@londi I'm assuming that you're reading this frame of the CAN bus, and it's not an OBD request.

To know what this frame represents you need to find a DBC file specific for your car. The [tool](https://www.csselectronics.com/pages/obd2-pid-table-on-board-diagnostics-j1979) you mentioned is for OBD protocol request response.

> If I send a 2016 tx frame via mqtt, the meatpi does not responde with a 2024 frame. (As explained in the docs)

it's possible that your car doesn't support this OBD request.

**Reply by londi** [reply]
@meatpiHQ thank you!

To be sure which sensors are available I tried the app infocar and find out, that not all sensors are available before starting the engine. 

I have found a dbc file for my car (REF file). Is that the file I need? How can I open and use it? (Sorry, I'm a newby on this topic)

Other question: What mqtt broker have you used for your tests? (E.g. mosquitto?) Just to be sure the problem is not depending on that.

Thank you in advance :-)

**Reply by meatpiHQ** [reply]
@londi That's great. You can load the DBC file in BUSMaster or savvyCAN so you can view the sensors values in real time.

You can lookup DBC file format, the DBC file will tell you what each CAN ID represents.

I use the MQTT broker in Home assistance, I think it's  mosquitto based.
